eclipse-collections-api/src/main/java/org/eclipse/collections/api/collection/ImmutableCollection.java

```diff
@@ -190,4 +196,37 @@
             Function<? super T, ? extends K> groupBy,
             Function0<? extends V> zeroValueFactory,
             Function2<? super V, ? super T, ? extends V> nonMutatingAggregator);
+
+    /**
+     * @since 9.0
+     */
+    default Stream<T> stream()
+    {
+        return StreamSupport.stream(this.spliterator(), false);
+    }
+
+    /**
+     * @since 9.0
+     */
+    default Stream<T> parallelStream()
+    {
+        return StreamSupport.stream(this.spliterator(), true);
+    }
+
+    /**
+     * @since 9.0
+     */
+    @Override
+    default Spliterator<T> spliterator()
+    {
+        return Spliterators.spliterator(this.iterator(), (long) this.size(), 0);
+    }
+
+    /**
+     * @since 9.0
+     */
+    default Collection<T> castToCollection()
+    {
+        return (Collection<T>) this;
```

Comment:
This can be made abstract or default and nothing will break.  Now that we can use default methods, I find myself wanting to use them where we can when we add new APIs as it will reduce the number of places client code would have to be fixed if these interfaces are implemented.